### PR TITLE
Move the overview task list to the welcome greeting to improve visibility

### LINKED ISSUE #6382: Move `Things to do` task list to `Welcome` card on Payments → Overview
Part of #6353

The **Payments → Overview** `Things to do` task list should be displayed as part of the `Welcome` greeting card.

* Move the existing task list inside the `Welcome` card.
* The existing `Things to do` header will be removed, now replaced by the `Welcome` greeting.
* When there are task list items to display, the `Welcome` and `Balances` cards should be split into separate cards
	* Welcome + Task List
	* Balances
* When there are **no** task list items to display, the `Welcome` & `Balances` cards should display as a single card.
	* Welcome + Balances

**Render the Welcome card with the task list when there are tasks to display:**

image

**Render the Welcome card merged with the Balances card when there are no tasks to display:**

image

See Slack discussion: p1685059256012029-slack-C03LA109BKM

### PR DESCRIPTION
**Pull request by Jinksi**
Fixes #6382 

#### Changes proposed in this Pull Request

This PR moved the Payments → Overview `Things to do` task list to be rendered under the `Welcome` greeting to improve visibility of tasks.

* When there are tasks to be displayed, the Welcome card is split from the AccountBalances card.
* When there are no tasks to be displayed, the Welcome & AccountBalances cards are displayed as a single card.

#### Testing instructions

* View the Payments → Overview screen with no disputes.
	* The Welcome/Balances card should not be split.
	* The TaskList should not be visible.

   	image

* View the Payments → Overview with 1+ active disputes.
	* The Welcome card should be split from the Balances card.
	* The TaskList should be visible under the Welcome card header with a dispute resolution task.
	* The `Things to do` task list should not be visible elsewhere on the Overview page.
	
   	image

## TODO

- [x] [Confirm which tasks are moving](https://github.com/Automattic/woocommerce-payments/pull/6397#issuecomment-1577518516) / if we are splitting the `Things to do` task list into 2 lists and including only urgent tasks in the Welcome banner.
	- Confirmed: **all existing tasks should appear in the welcome banner header**, so merchants know where can find any tasks. It would be confusing to have different tasks in different places. 

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [x] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: client/overview/style.scss [hunk 1]
```diff
@@ -9,17 +9,11 @@
 }
 
 // Adapted from woocommerce-admin:client/homescreen/style.scss
-.components-card.woocommerce-task-card {
```
Comment (Jinksi) [inline client/overview/style.scss:12]: Removed CSS related to redundant CardBody wrap

File: client/overview/task-list/index.js [hunk 2]
```diff
@@ -166,73 +158,50 @@ const TaskList = ( { overviewTasksVisibility, tasks } ) => {
 		return <div></div>;
 	}
 
-	const pendingTaskCount = visibleTasks.filter( ( task ) => ! task.completed )
-		.length;
 	return (
-		<Card
```
Comment (Jinksi) [inline client/overview/task-list/index.js:172]: Removed the Card, CardHeader and CardBody to allow this component to be placed inside a Card alongside the Welcome CardHeader

File: client/overview/index.js [hunk 3]
```diff
@@ -144,27 +147,35 @@ const OverviewPage = () => {
...
 					<>
-						<Card>
-							<Welcome />
-							<AccountBalances />
-						</Card>
+						{ showTaskList ? (
+							<>
+								<Card>
+									<Welcome />
+									<ErrorBoundary>
```
Comment (Jinksi) [inline client/overview/index.js:153]: I've wrapped the TaskList in an ErrorBoundary to allow the <Welcome /> component to render in the case of a TaskList error.

File: client/overview/index.js [hunk 4]
```diff
@@ -72,7 +72,10 @@ const OverviewPage = () => {
 	} );
 	const tasks =
 		Array.isArray( tasksUnsorted ) && tasksUnsorted.sort( taskSort );
+
 	const queryParams = getQuery();
+	const accountRejected =
+		accountStatus.status && accountStatus.status.startsWith( 'rejected' );
```
Comment (Jinksi) [inline client/overview/index.js:78]: No change to this variable, just moved above the `show...` variables to follow some sort of order to these declared vars.

File: client/overview/task-list/index.js [hunk 5]
```diff
@@ -5,14 +5,6 @@
...
 import React from 'react';
 import { __ } from '@wordpress/i18n';
-import {
-	Card,
-	CardBody,
-	CardHeader,
-	FlexItem,
-	FlexBlock,
-} from '@wordpress/components';
-import { Badge } from '@woocommerce/components';
```
Comment (Jinksi) [inline client/overview/task-list/index.js:15]: Badge was only used for the `Things to do` header, which has been removed.

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6397` or branch name `fix/6382-move-overview-task-list` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=fix/6382-move-overview-task-list) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 314fb8ab38d5029363ae1e636c20ad95bbca4ba0
- Build time: 2023-06-08 23:49:16 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by shendy-a8c** [comment]
Resolved merge conflict in `client/overview/index.js` with [bdc006f](https://github.com/Automattic/woocommerce-payments/pull/6397/commits/bdc006fc4c87914899ce1e4918cd265df3ab53f8).

Please help review that I did that correctly. Make sure your change and change from [this PR](https://github.com/Automattic/woocommerce-payments/pull/6414/files#diff-a70fc7fd837200557381243b7777d7df8134f8c183630fae8d73cf221a83f511) is merged correctly in `client/overview/index.js`.

**Comment by shendy-a8c** [comment]
https://github.com/Automattic/woocommerce-payments/assets/73803630/b2c4bd94-6efd-4469-8f37-186e51371bee

Couple things I like to highlight from my screen recording:

1. If I am initially at the Overview page, then there is new dispute, I then open the disputes list page where I can see the new dispute needing action, clicking the Payments > Overview menu won't reload the page, so the task isn't displayed there, which I think it's OK. Just pointing out my observation.

2. From the recording, the dispute is not displayed even though the greeting card and the balance card are separated from each other. I think that's because I dismissed previous task from the three dots menu on the right of the task. After doing that, even though there is new dispute, I can't find a way to make the task appear again. It should as it should be considered as a new task.

3. If I dismiss a task, should the greeting card and the balance card combine into 1 again? I don't think so. Current behavior: the 2 cards aren't combined even though visually, there is nothing below the greeting card (no task) because dismissed. I think current behavior is fine.


I think point number 2 is important to fix. What do you think?

I'm still in the process of reviewing the code diff.

**Comment by Jinksi** [comment]
Thanks for testing this and sharing a screen recording @shendy-a8c 🙌 

For (1): we could do an API fetch and refresh the active disputes cache each time the user visits the overview screen. This could be handled in #6371 by setting the `dateBefore` to the current datetime when calling `useDisputes()`. What do you think?

For (2): where the dispute task won't appear if it has been dismissed. We can fix this by not allowing the dispute task to be dismissed (the current implementation in #6371). I think this will fix the issue.

For (3): dismissed tasks leaving the Welcome/Balance cards split: perhaps this is OK as you mention, since it may be jarring for the cards to join after clicking dismiss (non-dispute tasks).

Another thing you've made me realise that we should address: the [`dispute-resolution-task`](https://github.com/Automattic/woocommerce-payments/blob/bdc006fc4c87914899ce1e4918cd265df3ab53f8/client/overview/task-list/tasks.js#L182) task may have been previously dismissed by the merchant, so we should use a new key to ensure it isn't filtered out as a dismissed task.

**Comment by haszari** [comment]
> This PR moved the Payments → Overview Things to do task list to be rendered under the Welcome greeting to improve visibility of important tasks.

@Jinksi What tasks are included in the elevated list attached to the `Welcome` card? The intention was only to show urgent / critical tasks there, not all the things in `Things to do` list. Please clarify what's moving and what's staying elsewhere (lower priority and lower on page) in the PR description and title.

**Comment by Jinksi** [comment]
> The intention was only to show urgent / critical tasks there, not all the things in Things to do list.

Ah, thanks @haszari. It'd be good to clarify this, since I did think the intention was to move the entire list based on this slack thread p1685128201119589/1685059256.012029-slack-C03LA109BKM:

> The banner notice is the Things to do tasklist, however this update removes the card header with “Things to do” in favor of placing the notice within the greeting header.
> @elizaan36 

> Please clarify what's moving

The current implementation  of this PR moves all tasks, so I'll start compile a list of tasks that are displayed in this list.

I'll mark this PR as `in progress` while I investigate.

**Comment by haszari** [comment]
> The current implementation of this PR moves all tasks, so I'll start compile a list of tasks that are displayed in this list.

Thanks @Jinksi – and apologies for not making this clear! The Figma has some good discussion (in comments) of which tasks and what "important" means in this context – let me know if you can't find that. From memory I think the only relevant task to move is `You have active dispute(s)`.

**Comment by Jinksi** [comment]
> The intention was only to show urgent / critical tasks there, not all the things in Things to do list.

@haszari I'll paste the clarification about this as we have discussed via slack: p1686002501578689/1685059256.012029-slack-C03LA109BKM.

**All existing tasks should appear in the welcome banner header**, so merchants know where can find any tasks. It would be confusing to have different tasks in different places.

**Review by shendy-a8c — APPROVED** [review]
Looks good!

Do you think this PR is OK to be merged independently from other related PRs? I am leaning toward yes.

Also, you brought up a good point here:
> Another thing you've made me realise that we should address: the [dispute-resolution-task](https://github.com/Automattic/woocommerce-payments/blob/bdc006fc4c87914899ce1e4918cd265df3ab53f8/client/overview/task-list/tasks.js#L182) task may have been previously dismissed by the merchant, so we should use a new key to ensure it isn't filtered out as a dismissed task.

Even after merging this PR's branch and https://github.com/Automattic/woocommerce-payments/pull/6371's into a local branch I created, I still couldn't bring back the task I dismissed, so even though there is a new dispute, merchant who dismissed the task won't be able to see new disputes.

I find that critical, so I created https://github.com/Automattic/woocommerce-payments/issues/6485.

**Comment by Jinksi** [comment]
> Do you think this PR is OK to be merged independently from other related PRs? I am leaning toward yes.

@shendy-a8c Yes I think so: this works as an incremental improvement in isolation of other PRs. Although I am confident we will also see the remaining Dispute task list improvements (#6371) merged in 6.1.0.

Merging!

**Pull request merged**